File: src/nl/hannahsten/texifyidea/structure/filter/LabelFilter.java
package nl.hannahsten.texifyidea.structure.filter;

import com.intellij.ide.util.treeView.smartTree.ActionPresentation;
import com.intellij.ide.util.treeView.smartTree.Filter;
import com.intellij.ide.util.treeView.smartTree.TreeElement;
import nl.hannahsten.texifyidea.TexifyIcons;
import nl.hannahsten.texifyidea.structure.latex.LatexStructureViewCommandElement;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

/**
 * @author Hannah Schellekens
 */
public class LabelFilter implements Filter {

    @Override
    public boolean isVisible(TreeElement treeElement) {
        if (!(treeElement instanceof LatexStructureViewCommandElement)) {
            return true;
        }

        LatexStructureViewCommandElement element = (LatexStructureViewCommandElement)treeElement;
        return !element.getCommandName().equals("\\label");
    }

    @Override
    public boolean isReverted() {
        return true;
    }

    @NotNull
    @Override
    public ActionPresentation getPresentation() {
        return LatexLabelFilterPresentation.INSTANCE;
    }

    @NotNull
    @Override
    public String getName() {
        return "latex.texify.filter.label";
    }

    /**
     * @author Hannah Schellekens
     */
    private static class LatexLabelFilterPresentation implements ActionPresentation {

        private static final LatexLabelFilterPresentation INSTANCE = new LatexLabelFilterPresentation();

        @NotNull
        @Override
        public String getText() {
            return "Show Labels";
        }

        @Override
        public String getDescription() {
            return "Show Labels";
        }

        @Override
        public Icon getIcon() {
            return TexifyIcons.DOT_LABEL;
        }
    }
}

File: src/nl/hannahsten/texifyidea/structure/filter/LabelFilter.kt
package nl.hannahsten.texifyidea.structure.filter

import com.intellij.ide.util.FileStructureFilter
import com.intellij.ide.util.treeView.smartTree.ActionPresentation
import com.intellij.ide.util.treeView.smartTree.TreeElement
import com.intellij.openapi.actionSystem.Shortcut
import nl.hannahsten.texifyidea.TexifyIcons
import nl.hannahsten.texifyidea.structure.latex.LatexStructureViewCommandElement
import nl.hannahsten.texifyidea.util.magic.CommandMagic
import javax.swing.Icon

/**
 * @author Hannah Schellekens
 */
class LabelFilter : FileStructureFilter {

    override fun isVisible(treeElement: TreeElement): Boolean {
        if (treeElement !is LatexStructureViewCommandElement) {
            return true
        }
        return !CommandMagic.labels.contains(treeElement.commandName)
    }

    override fun isReverted(): Boolean = true

    override fun getPresentation(): ActionPresentation = LatexLabelFilterPresentation.INSTANCE

    override fun getName(): String = "latex.texify.filter.label"

    override fun getCheckBoxText() = "Label commands"

    override fun getShortcut() = emptyArray<Shortcut>()

    /**
     * @author Hannah Schellekens
     */
    private class LatexLabelFilterPresentation : ActionPresentation {

        override fun getText(): String = "Show Labels"

        override fun getDescription(): String = "Show labels"

        override fun getIcon(): Icon = TexifyIcons.DOT_LABEL

        companion object {

            val INSTANCE = LatexLabelFilterPresentation()
        }
    }
}